src/nginx/t/config_rollouts_managed.t

```diff
@@ -0,0 +1,422 @@
+  'Service management socket ready.' );
+$t->run();
+
+################################################################################
+# send the first reqeust 
+my $response = ApiManager::http_get($NginxPort, '/shelves?key=this-is-an-api-key' );
+
+# verify the response
+my ( $response_headers, $response_body ) = split /\r\n\r\n/, $response, 2;
+like( $response_headers, qr/HTTP\/1\.1 200 OK/, 'Returned HTTP 200.' );
+is( $response_body, <<'EOF', 'Shelves returned in the response body.' );
+{ "shelves": [
+    { "name": "shelves/1", "theme": "Fiction" },
+    { "name": "shelves/2", "theme": "Fantasy" }
+  ]
+}
+EOF
+
+
+# waiting for the first report request is done
+is($t->waitforfile($t->{_testdir}."/".${report_done} . ".0"), 1, 'Report body file ready.');
+
+# waiting for the rolllout update
+is($t->waitforfile("$t->{_testdir}/${rollout_done}"), 1, 'Rollout body file ready.');
+
+# send the second request
+$response = ApiManager::http_get($NginxPort, '/shelves?key=this-is-an-api-key' );
+# verify the second response
+( $response_headers, $response_body ) = split /\r\n\r\n/, $response, 2;
+like( $response_headers, qr/HTTP\/1\.1 200 OK/, 'Returned HTTP 200.' );
+is( $response_body, <<'EOF', 'Shelves returned in the response body.' );
+{ "shelves": [
+    { "name": "shelves/1", "theme": "Fiction" },
+    { "name": "shelves/2", "theme": "Fantasy" }
+  ]
+}
+EOF
+
+# waiting for the second report request is done
+is($t->waitforfile($t->{_testdir}."/".${report_done} . ".1"), 1, 'Report body file ready.');
+
+$t->stop_daemons();
+
+
+# Service management requests verification
+my @servicemanagement_requests = ApiManager::read_http_stream( $t, 'servicemanagement.log' );
+# service config rollout requests are made by timer.
+# first two requests contain rollout information. Requests will be ignore
+# because of same rollout_id from the third.
+ok( scalar @servicemanagement_requests >= 2, 'Service management received two requests' );
+
+my $r = shift @servicemanagement_requests;
+is( $r->{verb}, 'GET', 'Rollout request was a get' );
+is( $r->{uri}, '/v1/services/endpoints-test.cloudendpointsapis.com/rollouts?filter=status=SUCCESS',
+  'Rollout request uri is correct' );
+is( $r->{headers}->{host}, "127.0.0.1:${ServiceManagementPort}", 'Host header was set' );
+
+$r = shift @servicemanagement_requests;
+is( $r->{verb}, 'GET', 'Service download request was a get' );
+is( $r->{uri}, '/v1/services/endpoints-test.cloudendpointsapis.com/configs/2016-08-25r3',
+  'Service config uri' );
+is( $r->{headers}->{host}, "127.0.0.1:${ServiceManagementPort}", 'Host header was set' );
+
+# validate rests of request has no change
+my $check_error_rest = 0;
+while($r = shift @servicemanagement_requests) {
+  if ($r->{verb} ne 'GET' ||
+    $r->{uri} ne '/v1/services/endpoints-test.cloudendpointsapis.com/rollouts?filter=status=SUCCESS' ||
+    $r->{headers}->{host} ne "127.0.0.1:${ServiceManagementPort}" ) {
```

Comment:
line 206 is not needed.  servicemanagement_requests is read from servicemanagement.log file which was wrotten by that http server.  Its "host" has to be that port.